### PR TITLE
Clarify the effect of PolygonCollection properties on Quiver

### LINKED ISSUE #20298: plt.quiver linestyle option doesn't work?.....
### Bug report

**Bug summary**
**I set the linestyle option of plt.quiver, but a straight line always comes out.**  
Did you write the wrong code? Or do I need to change the Matplotlib Version?

**Code for reproduction**

```python
def draw(x, y):
    xPlusy = ((x[0]+y[0])/2,(x[1]+y[1])/2)
    array = np.array([[0, 0, x[0], x[1]],
                      [0, 0, y[0], y[1]],
                      [0, 0, xPlusy[0], xPlusy[1]]])
   
    X, Y, U, V = zip(*array)
    print("X =",X)
    print("Y =",Y)
    print("U =",U)
    print("V =",V)
    plt.figure()
    plt.figure(figsize=(16,10))
    plt.ylabel('Y-axis')
    plt.xlabel('X-axis')
    ax = plt.gca()
    ax.quiver(X, Y, U, V, angles='xy', scale_units='xy',color=['b','r','g'],scale=1, linestyle=':')
    ax.set_xlim([-4, 6])
    ax.set_ylim([-1, 5])
    # ax.axis('off')
    # plt.draw()
    plt.show()

a = np.array([-3, 0])
b = np.array([4, 4])

draw(a, b)
```

**Actual outcome**
image

**Matplotlib version**

  * Operating system: MAC or Linux(Both same issue....)
image

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Closes #20298.

### CODE REVIEW

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -188,7 +193,7 @@
 
 Returns
 -------
-matplotlib.quiver.Quiver
+quiver : `~matplotlib.quiver.Quiver`
```
Comment (anntzer) [inline lib/matplotlib/quiver.py:196]: Can you just put the quoted type name here?

Comment (timhoffm) [reply lib/matplotlib/quiver.py]: Yes. I thought there were some issues in numpydoc with that, but it seems to work now.

**Review by anntzer — APPROVED** [review]

**Comment by jklymak** [comment]
```
/home/circleci/project/lib/matplotlib/axes/_axes.py:docstring of matplotlib.axes._axes.Axes.quiver:11: WARNING: py:obj reference target not found: PolygonCollection
/home/circleci/project/lib/matplotlib/pyplot.py:docstring of matplotlib.pyplot.quiver:11: WARNING: py:obj reference target not found: PolygonCollection
/home/circleci/project/lib/matplotlib/quiver.py:docstring of matplotlib.quiver.Quiver:48: WARNING: py:obj reference target not found: PolygonCollection
```

**Review by jklymak — APPROVED** [review]
Feel free to self merge when CI passes....

**Pull request merged**